stdlib/GMP_jll/src/GMP_jll.jl

```diff
@@ -4,49 +4,53 @@
 baremodule GMP_jll
 using Base, Libdl
 
-const PATH_list = String[]
-const LIBPATH_list = String[]
-
 export libgmp, libgmpxx
 
 # These get calculated in __init__()
 const PATH = Ref("")
+const PATH_list = String[]
 const LIBPATH = Ref("")
+const LIBPATH_list = String[]
 artifact_dir::String = ""
-libgmp_handle::Ptr{Cvoid} = C_NULL
 libgmp_path::String = ""
-libgmpxx_handle::Ptr{Cvoid} = C_NULL
 libgmpxx_path::String = ""
 
+_libgmp_dependencies = LazyLibrary[]
+_libgmpxx_dependencies = LazyLibrary[]
+
 if Sys.iswindows()
-    const libgmp = "libgmp-10.dll"
-    const libgmpxx = "libgmpxx-4.dll"
+    const _libgmp_path = BundledLazyLibraryPath("libgmp-10.dll")
+    const _libgmpxx_path = BundledLazyLibraryPath("libgmpxx-4.dll")
 elseif Sys.isapple()
-    const libgmp = "@rpath/libgmp.10.dylib"
-    const libgmpxx = "@rpath/libgmpxx.4.dylib"
+    const _libgmp_path = BundledLazyLibraryPath("libgmp.10.dylib")
+    const _libgmpxx_path = BundledLazyLibraryPath("libgmpxx.4.dylib")
 else
-    const libgmp = "libgmp.so.10"
-    const libgmpxx = "libgmpxx.so.4"
+    const _libgmp_path = BundledLazyLibraryPath("libgmp.so.10")
+    const _libgmpxx_path = BundledLazyLibraryPath("libgmpxx.so.4")
 end
 
+const libgmp = LazyLibrary(
+    _libgmp_path,
+    dependencies=_libgmp_dependencies,
+)
+
+const libgmpxx = LazyLibrary(
+    _libgmpxx_path,
+    dependencies=_libgmpxx_dependencies,
```

Comment:
```suggestion
    dependencies=[libgmp, libstdcxx, libgcc_s],
```

Turns out `libgmpxx` has a bunch of dependencies that we didn't really capture previously.